### PR TITLE
[CI] [Hotfix] Add an apt-update step to the ubuntu CI jobs.

### PR DESCRIPTION
**Pull request by NlightNFotis**
This stops it from getting an HTTP 404 while downloading
dependencies if they have been updated since the image
came out.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/pull-request-checks.yaml [hunk 1]
```diff
@@ -365,7 +365,9 @@ jobs:
         with:
           submodules: recursive
       - name: Download test dependencies
-        run: sudo apt install openjdk-11-jdk-headless
+        run: |
+          sudo apt update
+          sudo apt install openjdk-11-jdk-headless
```
Comment (hannes-steffenhagen-diffblue) [inline .github/workflows/pull-request-checks.yaml:375]: Should our other jobs get the same treatment?

Comment (NlightNFotis) [reply .github/workflows/pull-request-checks.yaml:375]: Probably.

Comment (NlightNFotis) [reply .github/workflows/pull-request-checks.yaml:375]: @hannes-steffenhagen-diffblue Okay I applied it to all the jobs.

**Review by piotr-grabalski — APPROVED** [review]
looks quite straightforward

**Pull request merged**